src/transformers/trainer.py

```diff
@@ -2711,6 +2724,38 @@ def opt_load_hook(mod, opt):
                     self.lr_scheduler.load_state_dict(torch.load(os.path.join(checkpoint, SCHEDULER_NAME)))
                 reissue_pt_warnings(caught_warnings)
 
+    def _load_callback_state(self):
+        """If callback states exist and were passed in, restore their states if enabled"""
+        if not self.args.restore_callback_states_from_checkpoint:
+            return
+        # Callback states are stored in stateful_callbacks
+        not_found = []
+        for stored_callback, data in self.state.stateful_callbacks.items():
```

Comment:
Can we ever have more than one callback of the same type e.g. different metric loggers? Only concern with logic below is that we might load state from one class into another

Reply:
Good call Amy, just in case we can, I've gone ahead and added logic in + a test to verify :)